site/tutorials/snippets/17/quality-gates/setupQualityGate.md

```diff
@@ -0,0 +1,187 @@
+
+    <!-- command -->
+    ```
+    keptn trigger delivery --project=sockshop --service=carts --image=docker.io/keptnexamples/carts:0.13.2
+    ```
+    
+    <!-- bash 
+    verify_test_step $? "trigger delivery for carts failed" 
+    wait_for_deployment_with_image_in_namespace "carts" "sockshop-staging" "docker.io/keptnexamples/carts:0.13.2"
+    verify_test_step $? "Deployment carts not available, exiting..."
+    echo "Waiting for a little bit!"
+    wait_for_event_with_field_output "sh.keptn.event.release.finished" ".data.result" "fail" "sockshop"
+    sleep 60
+    -->
+
+1. Verify that the slow build has reached your `dev` and `staging` environments by opening a browser for both environments. This may take 5 to 10 minutes. Get the URLs with these commands:
+
+    <!-- command -->
+    ```
+    echo http://carts.sockshop-dev.$(kubectl -n keptn get ingress api-keptn-ingress -ojsonpath='{.spec.rules[0].host}')
+    ```
+
+    <!-- command -->
+    ```
+    echo http://carts.sockshop-staging.$(kubectl -n keptn get ingress api-keptn-ingress -ojsonpath='{.spec.rules[0].host}')
+    ```
+
+
+![carts service](./assets/carts-dev-2.png)
+
+![carts service](./assets/carts-staging-2.png)
+
+
+## Quality gate in action
+Duration: 7:00 
+
+After triggering the deployment of the carts service in version v0.13.2, the following status is expected:
+
+* **Dev stage:** The new version is deployed in the dev stage and the functional tests passed.
+  * To verify, open a browser and navigate to: 
+  <!-- command -->
+  ```
+  echo http://carts.sockshop-dev.$(kubectl -n keptn get ingress api-keptn-ingress -ojsonpath='{.spec.rules[0].host}')
+  ```
+
+* **Staging stage:** In this stage, version v0.13.2 will be deployed and the performance test starts to run for about 10 minutes. After the test is completed, Keptn triggers the test evaluation and identifies the slowdown. Consequently, a roll-back to version v0.13.1 in this stage is conducted and the promotion to production is not triggered.
+
+
+* **Production stage:** The slow version is **not promoted** to the production stage because of the active quality gate in place. Thus, still version v0.13.1 is expected to be in production.
+  * To verify, navigate to: 
+  <!-- command -->
+  ```
+  echo http://carts.sockshop-production.$(kubectl -n keptn get ingress api-keptn-ingress -ojsonpath='{.spec.rules[0].host}')
+  ```
+
+## Verify the quality gate in Keptn's Bridge
+Duration: 3:00
+
+Take a look in the Keptn's bridge and navigate to the last deployment. You will find a quality gate evaluation that got a `fail` result when evaluation the SLOs of our carts microservice. Thanks to this quality gate the slow build won't be promoted to production but instead automatically rolled back.
+
+To verify, the [Keptn's Bridge](https://keptn.sh/docs/0.14.x/reference/bridge/) shows the deployment of v0.13.2 and then the failed test in staging including the roll-back.
```

Comment:
```suggestion
To verify, the [Keptn's Bridge](https://keptn.sh/docs/0.17.x/reference/bridge/) shows the deployment of v0.13.2 and then the failed test in staging including the roll-back.
```

Reply:
The link was already incorrect for Keptn version 0.14. I fixed it for both versions now.